internal/controller/pkg/manager/reconciler.go

```diff
@@ -564,6 +551,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	return pullBasedRequeue(p.GetPackagePullPolicy()), errors.Wrap(r.client.Status().Update(ctx, p), errUpdateStatus)
 }
 
+func packageHealth(pr v1.PackageRevision) xpv1.Condition {
+	revisionHealth := pr.GetCondition(v1.TypeRevisionHealthy)
+	runtimeHealth := pr.GetCondition(v1.TypeRuntimeHealthy)
+
+	revisionHealthy := revisionHealth.Status == corev1.ConditionTrue
+	runtimeHealthy := runtimeHealth.Status == corev1.ConditionTrue
+	if _, hasRuntime := pr.(v1.PackageRevisionWithRuntime); !hasRuntime {
+		// If the package revision does not have a runtime, we skip checking the runtime health.
+		runtimeHealthy = true
+	}
+
+	if revisionHealthy && runtimeHealthy {
+		return v1.Healthy()
+	}
+
+	if !revisionHealthy {
+		m := fmt.Sprintf("Package revision health is %q", revisionHealth.Status)
+		if revisionHealth.Message != "" {
+			m += " with message: " + revisionHealth.Message
+		}
+		return v1.Unhealthy().WithMessage(m)
```

Comment:
Why add this message? If the user's reading this status condition message, won't they also see the `RevisionHealthy` status condition message right next to it? Seems a bit redundant to me.